tests/std/tests/GH_002030_asan_annotate_string/test.cpp

```diff
@@ -287,6 +287,94 @@ STATIC_ASSERT(_Container_allocation_minimum_asan_alignment<
                   basic_string<wchar_t, char_traits<wchar_t>, implicit_allocator<wchar_t>>>
               == 2);
 
+// Helper class for `ArenaAllocator`, where data is linearly allocated in a finite buffer.
+class Arena {

```

Comment:
@CaseyCarter - I have this `Arena` and `ArenaAllocator` classes duplicated in the `basic_string` and `vector` test files. I'd love to refactor it into a shared location so that it can be imported in both tests.

Do you all have some kind of "shared test utils" header in this repo or do we prefer to keep the tests fully self-contained?

Reply:
`tests/std/include` is where shared test headers go.

Reply:
Thanks. Since we've greatly simplified the test case, I don't think I'll need to refactor any classes into a shared location, but I'll remember this for the future :) . _Resolving_